### PR TITLE
pcie_hotplug_opt: Add support for aarch64 platform

### PR DESCRIPTION
**Pull request by zhenyzha**
pcie_hotplug_opt add support for aarch64 platform

ID:1940270
Signed-off-by: Zhenyu Zhang <zhenyzha@redhat.com>

### CODE REVIEW

**Comment by zhenyzha** [comment]
```
 (1/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.aarch64.io-github-autotest-qemu.pci_bridge.basic.one_bridge.arm64-pci: PASS (66.24 s)
 (2/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.aarch64.io-github-autotest-qemu.pci_bridge.basic.multi_parallel_bridges.arm64-pci: PASS (63.71 s)
 (3/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.aarch64.io-github-autotest-qemu.pci_bridge.with_migration.one_bridge.arm64-pci: PASS (102.88 s)
 (4/4) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.aarch64.io-github-autotest-qemu.pci_bridge.with_migration.multi_parallel_bridges.arm64-pci: PASS (103.07 s)

```

@PaulYuuu Could you pls help review, thanks.

File: qemu/tests/cfg/pci_bridge.cfg [hunk 1]
```diff
@@ -2,7 +2,7 @@
     virt_test_type = qemu
     login_timeout = 360
     type = pci_bridge
-    only i440fx pseries
+    only i440fx pseries aarch64
```
Comment (PaulYuuu) [inline qemu/tests/cfg/pci_bridge.cfg:5]: i440FX and pseries use `PCI` bridge, but `aarch64` uses `PCIE` bridge, so this cfg is not applicable to aarch64.

Comment (zhenyzha) [reply qemu/tests/cfg/pci_bridge.cfg:5]: You can see that aarch64 supports this device, and it works in cmd.
```
[root@ampere-hr330a-13 tests]# /usr/libexec/qemu-kvm -device pci-bridge,?
pci-bridge options:
  addr=<int32>           - Slot and optional function number, example: 06.0 or 06 (default: -1)
  bus-reserve=<uint32>   -  (default: 4294967295)
  chassis_nr=<uint8>     -  (default: 0)

```

```
    -device pcie-root-port,id=pcie-root-port-1,port=0x1,addr=0x1.0x1,bus=pcie.0,chassis=2 \
    -device virtio-gpu-pci,bus=pcie-root-port-1,addr=0x0 \
    -device pci-bridge,id=pci_bridge0,bus=pcie.0,addr=0x2,chassis_nr=1 \
```
Of course, we only test from pcie-root-port when testing hotplug, so I banned related cases. But I think it is possible to bring this -device to boot and migrate.

What do you think?

Comment (PaulYuuu) [reply qemu/tests/cfg/pci_bridge.cfg:5]: Not exactly. `q35` also supports pci-bridge, but we wouldn't test it, so you can see another test scenario named `pcie_controllers.cfg` for `q35`.

Comment (zhenyzha) [reply qemu/tests/cfg/pci_bridge.cfg:5]: Thanks a lot, I agree with you, I will investigate `pcie_controllers.cfg` next.

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.pcie_hotplug_opt.arm64-pci: PASS (129.66 s)

@PaulYuuu Could you pls help review, thanks.

File: qemu/tests/cfg/pcie_hotplug_opt.cfg [hunk 2]
```diff
@@ -1,5 +1,5 @@
 - pcie_hotplug_opt:
-    only q35
+    only q35 aarch64
```
Comment (PaulYuuu) [inline qemu/tests/cfg/pcie_hotplug_opt.cfg:2]: `q35` and `aarch64` are different levels, so suggest split into 2 lines.

Comment (zhenyzha) [reply qemu/tests/cfg/pcie_hotplug_opt.cfg]: Ok, corrected.

**Comment by zhenyzha** [comment]
@PaulYuuu Could you pls help review again, thanks.

**Comment by PaulYuuu** [comment]
> @PaulYuuu Sorry, I am a bit confused. If it is written in two lines, no case will pass the filter.

`q35` is a machine type of x86, so you can update the line to `only x86_64 aarch64` and `x86_64: only q35`

**Comment by zhenyzha** [comment]
> > @PaulYuuu Sorry, I am a bit confused. If it is written in two lines, no case will pass the filter.
> 
> `q35` is a machine type of x86, so you can update the line to `only x86_64 aarch64` and `x86_64: only q35`

Thanks a lot, corrected.

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.aarch64.io-github-autotest-qemu.pcie_hotplug_opt.arm64-pci: PASS (61.14 s)

@PaulYuuu Could you pls help review again, thanks.

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**